modules/apps/29-fee/keeper/msg_server_test.go

```diff
@@ -29,16 +30,25 @@ func (suite *KeeperTestSuite) TestRegisterCounterpartyAddress() {
 			true,
 			func() { counterparty = "arbitrary-string" },
 		},
+		{
+			"channel does not exist",
+			false,
+			func() { channelID = "channel-22" },
+		},
 	}
 
 	for _, tc := range testCases {
 		suite.SetupTest()
 		ctx := suite.chainA.GetContext()
+		suite.coordinator.Setup(suite.path) // setup channel
 
 		sender = suite.chainA.SenderAccount.GetAddress().String()
 		counterparty = suite.chainB.SenderAccount.GetAddress().String()
+		channelID = suite.path.EndpointA.ChannelID
+
 		tc.malleate()
-		msg := types.NewMsgRegisterCounterpartyAddress(sender, counterparty, ibctesting.FirstChannelID)
+
+		msg := types.NewMsgRegisterCounterpartyAddress(sender, counterparty, suite.path.EndpointA.ChannelConfig.PortID, channelID)
 
 		_, err := suite.chainA.SendMsgs(msg)
```

Comment:
For some reason, the test suite is exploding here and returning the error message we expect (channel does not exist) but it's panicking and exiting rather than being handled.

Reply:
`SendMsgs` assumes the message to pass and errors [out otherwise](https://github.com/cosmos/ibc-go/blob/bd086506f9256c4d5776bc38d2e930b523eb5125/testing/simapp/test_helpers.go#L261). We could another function to the testing package to send a message expected to fail

Reply:
Gotcha. I'll just call the function directly similar to the other tests